### ISSUE TITLE
config ignored for dhcp service

### ISSUE THREAD

**Issue by bryfry** [psychomario/PyPXE#116]
[These arguments need their order reversed](https://github.com/psychomario/PyPXE/blob/c52d87686a62c8ae0488074d470d87e8e2f0914b/pypxe-server.py#L74-L75).  In the current order, no matter what you put in the config the USE_DHCP will be false.

Additionally, the documented defaults do not match the code:
[Documented](https://github.com/psychomario/PyPXE/blame/c52d87686a62c8ae0488074d470d87e8e2f0914b/README.md#L49) / [Actual](https://github.com/psychomario/PyPXE/blob/c52d87686a62c8ae0488074d470d87e8e2f0914b/pypxe-server.py#L37)

This may be why the config file gets ignored.  As it currently stands you have to explicitly call `--dhcp`.  I'd prefer to use the config.

**Comment by bryfry** [comment]
Also the documented defaults do not match the code:
[Documented](https://github.com/psychomario/PyPXE/blame/c52d87686a62c8ae0488074d470d87e8e2f0914b/README.md#L49) / [Actual](https://github.com/psychomario/PyPXE/blob/c52d87686a62c8ae0488074d470d87e8e2f0914b/pypxe-server.py#L37)

This may be why the config file gets ignored.  As it currently stands you have to explicitly call `--dhcp`.  I'd prefer to use the config.

**Comment by mmattioli** [comment]
It is intentional to explicitly call `--dhcp`; there might already be an existing DHCP server on the network sending clients to PyPXE.

Good catch on the documentation discrepancy, thank you.

If using a configuration file, the arguments are overwritten when the config is loaded. This was addressed in #84.

**Comment by bryfry** [comment]
If I run PyPXE with the `example_cfg.json` file I should expect DHCP to startup since ["USE_DHCP" : true,](https://github.com/psychomario/PyPXE/blob/c52d87686a62c8ae0488074d470d87e8e2f0914b/example_cfg.json#L17).  Actual result is below:

```
$ sudo python pypxe-server.py --debug all --config example_cfg.json
2015-10-22 22:15:17,914 [INFO] PyPXE Starting TFTP server...
2015-10-22 22:15:17,914 [DEBUG] PyPXE.TFTP NOTICE: TFTP server started in debug mode. TFTP server is using the following:
2015-10-22 22:15:17,914 [DEBUG] PyPXE.TFTP Server IP: 0.0.0.0
2015-10-22 22:15:17,914 [DEBUG] PyPXE.TFTP Server Port: 69
2015-10-22 22:15:17,915 [DEBUG] PyPXE.TFTP Network Boot Directory: .
2015-10-22 22:15:17,915 [INFO] PyPXE PyPXE successfully initialized and running!
```

Given that the defaults are without the DHCP service I think the --no-dhcp should be removed.  This is also a fix for the issue of the config not being recognized because of the argument parsing order.

See pr #118

**Comment by mmattioli** [comment]
1. Looks like you opened 2 PRs, please only submit 1
2. You need to open a PR against the `development` branch so we can test before merging to `master`

Please take care of the aforementioned and we will review. Thanks.

**Comment by bryfry** [comment]
Closed the PR on master

**Comment by psychomario** [comment]
Fixed in #119
